### DISCUSSION TITLE
Keep Threshold line on screen

### DISCUSSION THREAD

**Discussion by pfortmann** [patrykandpatrick/vico#634] [Questions] [answered]
Helllo, I have a requirement to keep the threshold lines on screen, even if the values won't reach that threshold line.
Im using the Horizontal Line from Version 2.
For examples the ranges could be in ranges from 1..20, and my Threshold line is set to 30. Is there a way to always show the Threshold line? Expected behaviour for me is to "rescale" the graph.

**Comment by patrickmichalik** [comment]
Hello! You can use `AxisValueOverrider` to do this. I’d suggest using an extra for the `HorizontalLine`’s _y_ value and creating an `AxisValueOverrider` that ensures the maximum _y_ value is no smaller than the `HorizontalLine`’s _y_ value. Let me know if you have any questions.

**Reply by pfortmann** [reply]
Yes this works, but how could i ensure that the threshholds y value isnt below the normal values. On some graphs i have values between 0-35 and a threshold of 20 (that way the regular values would be cut off if the maxY was set to 20 by the valueOverrider). My plan was to check for maxY in the model, which has been removed in Version 2? Is there some way to get charts maxY value from CartesianChartModelProducer?

**Reply by patrickmichalik** [reply, marked as answer]
The data’s minimum and maximum _y_ values are passed to `AxisValueOverrider#getMaxY`, @pfortmann. I’d suggest using something along these lines:

```kt
override fun getMaxY(minY: Float, maxY: Float, extraStore: ExtraStore) =
    maxY.coerceAtLeast(extraStore[horizontalLineYKey])
```

**Reply by pfortmann** [reply]
Thank you a lot! This works

**Reply by patrickmichalik** [reply]
I’m glad to hear that. You’re welcome!
